add container name fields to custom flow transformer

orozery opened the pull request:
Hi @safchain 
In this PR we make a change to our `custom1` flow transformer to include container names (via the runc probe).

Thanks in advance!

hunchback commented:
run skydive-unit-tests

hunchback commented:
run skydive-functional-tests-backend-orientdb

hunchback commented:
run skydive-compile-tests

hunchback commented:
run skydive-compile-tests

orozery commented:
run skydive-cdd-overview-tests

safchain commented:
run skydive-functional-tests-backend-orientdb

safchain commented on contrib/objectstore/subscriber/secadvisor.go:
```diff
@@ -36,11 +40,44 @@ type SecurityAdvisorFlow struct {
+}
+
+func (ft *SecurityAdvisorFlowTransformer) queryContainerName(ipString string) (string, error) {
+	node, err := ft.gremlinClient.GetNode(g.G.V().Has("Runc.Hosts.IP", ipString))
```
It should be better to first find the interface having the NodeTID (meaning the capture node) from there find the related node with the IP. I'm saying that because just checking with the IP you may have conflict.

orozery commented:
@safchain added the TID to the gremlin query (but a bit differently, to avoid the case of an empty response).
Also made few more minor changes:
- added NodeType field
- changed format of Network and Transport fields

This hopefully is the last PR required by the current version of security advisor.
Thanks!

orozery commented:
run skydive-k8s-tests
run skydive-functional-tests-backend-orientdb

orozery commented:
@hunchback @safchain per your request, I've added unit tests for the entire code in this PR (and more).
Hoping this solves the last obstacle for merging this.

hunchback commented:
@orozery

we are lacking integration tests which use minio or the likes. See for example how k8s uses minikube and enables full integration locally or on the ci (w/o depending on an external k8s cluster).

And then there is a seperate question of if the implementation in its current form is generic enough ... which also means moving ibm specific into configuration ... (even that the price of a performance hit).

hunchback commented:
@safchain / @orozery / @eranra - I want to open for discussion how we can enable this code to be tested by community - to fully test we would need:
- k8s / containerd - from which we get topology info used for resolving flow.IP to a stable pod/container name
- s3 object store - to which we store our flows

My suggestion is that we extend `queryContainerName` so that it attempts either `runc` based gremlin (as is currently used) or `docker` based gremlin - enabling to test code also on older k8s/docker not just k8s/containerd.

Thus now we would depend for running integration / ci tests on:
- k8s/docker - which we get with our k8s testing setup script `scripts/ci/install-minikube.sh`
- s3 object store - which I suggest *should* be installed using a new script `scripts/ci/install-minio.sh` (I know we have minio in Ansible but I prefer an independent script).

And then minimal CI test should:
- create a k8s pod
-  inject a flow
- resolve IP to pod/container-name
- store to object-store
- *verify resulting object*

eranra commented:
Hi @hunchback queryContainerName extensions - not sure you see the entire set of requirements we have in that area .... need something more than just runc and docker, we need the ability to support cases in which the IP is a 'generic compute' so for example VM and or bare-metal  ... this extension is needed but doesn't have to be part of this PR, we can add this at later phase (and we will do that because we have to, part  of this will be changes to the code that will change from 'containers' to 'compute entity').

eranra commented:
Hi @hunchback  --> s3 object store - Totally understand the requirement to extend the testing capabilities and Skydive CI for Object storage. @safchain suggested that we will check this PR as-is, without making this significant change to the CI. I would like to do that and have another PR following this one to extend integration/functional/CI tests. If you can suggest additions or changes to the unit-test code it will be great for this PR + whatever manual tests that @safchain could not perform to verify that things work as expected. This should be sufficient to accept this PR. 

I totally agree that we need a PR that will check object storage ... not just this PR content ahtt adds A_Name/B_Name but also other (already accepted) PR's that write/read from object storage in json/stream etc. We should not postpone acceptance of this PR based on this ... but I totally agree we need to work *now* on the needed CI change to have proper object storage tests. This aligns well with what you told me about the thinking to move object-storage from being 'stand alone' / API.

So, correct, but let's make this a following PR and not block this PR from being accepted

hunchback commented:
https://github.com/skydive-project/skydive/pull/1703#issuecomment-483577483

@eranra / @orozery - ok fine - I will review

orozery commented:
run skydive-functional-tests-backend-orientdb

orozery commented:
run skydive-functional-tests-backend-orientdb

orozery commented:
run skydive-functional-tests-backend-orientdb

hunchback commented:
@orozery - please upload here a an example of the output objectstorge - so that it can be reviewed - this practice can be used until integration tests in the CI are implemented.

orozery commented:
@hunchback what do you mean by "output" ?

orozery commented:
BTW we have integration tests implemented.
You can see their output here:

http://ci-logs.skydive.network/skydive-functional-tests-backend-elasticsearch/builds/3370/log.html

```
=== RUN   Test_SecurityAdvisor
2019-04-18T08:10:02.723Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, capture, write ---> true
2019-04-18T08:10:02.735Z	DEBUG	client/client.go:357 (*OnDemandProbeClient).onAPIWatcherEvent	jenkins-slave-2.rdocloud: New watcher event set for 3cbd8c7b-f5ba-4918-6b1e-aa840860a9ab
2019-04-18T08:10:02.742Z	DEBUG	server/server.go:77 (*OnDemandProbeServer).registerProbe	jenkins-slave-2.rdocloud: Attempting to register probe on node cni0
2019-04-18T08:10:02.743Z	DEBUG	tests/tests.go:299 execCmds	jenkins-slave-2.rdocloud: Executing command [podman run -d --name secadvtest --ip=10.88.0.49 --hostname=myhost nginx]
2019-04-18T08:10:02.766Z	DEBUG	server/server.go:119 (*OnDemandProbeServer).registerProbe	jenkins-slave-2.rdocloud: New active probe on: {"ID":"3e08b8c7-0ef5-4fa5-5f57-59becb3c5847","Metadata":{"BusInfo":"N/A","Driver":"bridge","EncapType":"ether","FDB":[{"Flags":["NTF_SELF"],"MAC":"33:33:00:00:00:01","State":["NUD_PERMANENT"],"IfIndex":13327},{"Flags":["NTF_SELF"],"MAC":"33:33:00:00:00:02","State":["NUD_PERMANENT"],"IfIndex":13327},{"Flags":["NTF_SELF"],"MAC":"01:00:5e:00:00:01","State":["NUD_PERMANENT"],"IfIndex":13327},{"Flags":["NTF_SELF"],"MAC":"33:33:ff:74:72:4e","State":["NUD_PERMANENT"],"IfIndex":13327},{"Flags":["NTF_SELF"],"MAC":"33:33:ff:00:00:00","State":["NUD_PERMANENT"],"IfIndex":13327},{"MAC":"d6:0c:9b:74:72:4e","State":["NUD_PERMANENT"],"Vlan":1,"IfIndex":13327},{"MAC":"d6:0c:9b:74:72:4e","State":["NUD_PERMANENT"],"IfIndex":13327}],"IPV4":["10.88.0.1/16"],"IPV6":["fe80::d40c:9bff:fe74:724e/64"],"IfIndex":13327,"LinkFlags":["UP","BROADCAST"],"LinkNetNsID":0,"LinkNetNsName":"","MAC":"d6:0c:9b:74:72:4e","MTU":1500,"Metric":{"RxBytes":384,"RxPackets":6,"TxBytes":1036,"TxPackets":10},"Name":"cni0","Neighbors":[{"MAC":"33:33:ff:74:72:4e","IP":"ff02::1:ff74:724e","State":["NUD_NOARP"],"IfIndex":13327},{"MAC":"33:33:00:00:00:16","IP":"ff02::16","State":["NUD_NOARP"],"IfIndex":13327}],"RoutingTables":[{"ID":254,"Src":"10.88.0.1","Routes":[{"Protocol":2,"Prefix":"10.88.0.0/16","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":2,"Prefix":"fe80::/64","NextHops":[{"Priority":256,"IfIndex":13327}]}]},{"ID":255,"Src":"10.88.0.1","Routes":[{"Protocol":2,"Prefix":"10.88.0.0/32","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":2,"Prefix":"10.88.0.1/32","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":2,"Prefix":"10.88.255.255/32","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":2,"Prefix":"fe80::/128","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":2,"Prefix":"fe80::d40c:9bff:fe74:724e/128","NextHops":[{"Priority":0,"IfIndex":13327}]},{"Protocol":3,"Prefix":"ff00::/8","NextHops":[{"Priority":256,"IfIndex":13327}]}]}],"State":"DOWN","TID":"4037dc08-f384-5a64-4090-657e6dcc26ad","Type":"bridge"},"Host":"jenkins-slave-2.rdocloud","Origin":"agent.jenkins-slave-2.rdocloud","CreatedAt":1555574998894,"UpdatedAt":1555575002389,"DeletedAt":null,"Revision":12}(&{{3cbd8c7b-f5ba-4918-6b1e-aa840860a9ab} G.V().Has("Name", "cni0")     0  0 0 0 false false false L2 })
2019-04-18T08:10:02.782Z	DEBUG	client/client.go:83 (*OnDemandProbeClient).OnStructMessage	jenkins-slave-2.rdocloud: Capture start request succeeded Namespace OnDemand Type CaptureStartReply UUID c75da386-3341-4fa2-6bd6-dad260ad0c48 Status 200 Obj (246 bytes)
2019-04-18T08:10:02.969Z	DEBUG	netns/netns.go:134 (*Probe).Register	jenkins-slave-2.rdocloud: Register network namespace: /var/run/netns/cni-a2f256fe-90f6-f3b6-d616-b6230c4fc4de
2019-04-18T08:10:03.050Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.064Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555574993064, 1555575003064
2019-04-18T08:10:03.073Z	DEBUG	netns/netns.go:169 (*Probe).Register	jenkins-slave-2.rdocloud: Network namespace added: 3,4026533337
2019-04-18T08:10:03.078Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.095Z	INFO	probes/gopacket.go:158 (*GoPacketProbe).Run	jenkins-slave-2.rdocloud: AfPacket Capture started on cni0 with First layer: Ethernet
2019-04-18T08:10:03.124Z	DEBUG	netns/netns.go:187 (*Probe).Register	jenkins-slave-2.rdocloud: Registering namespace: 3,4026533337
2019-04-18T08:10:03.128Z	DEBUG	netlink/netlink.go:1033 (*NetNsProbe).start	jenkins-slave-2.rdocloud: Start polling netlink event for 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.130Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.128Z	DEBUG	netlink/netlink.go:830 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize Netlink interfaces for 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.132Z	DEBUG	netlink/netlink.go:840 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize ADD lo(1,device) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.133Z	DEBUG	netlink/netlink.go:569 (*NetNsProbe).addLinkToTopology	jenkins-slave-2.rdocloud: Unable get Bus Info from ethtool (lo): operation not supported
2019-04-18T08:10:03.138Z	DEBUG	netlink/netlink.go:840 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize ADD gre0(2,gre) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.139Z	DEBUG	netlink/netlink.go:569 (*NetNsProbe).addLinkToTopology	jenkins-slave-2.rdocloud: Unable get Bus Info from ethtool (gre0): operation not supported
2019-04-18T08:10:03.144Z	DEBUG	netlink/netlink.go:840 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize ADD gretap0(3,gretap) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.147Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.148Z	DEBUG	netlink/netlink.go:569 (*NetNsProbe).addLinkToTopology	jenkins-slave-2.rdocloud: Unable get Bus Info from ethtool (gretap0): operation not supported
2019-04-18T08:10:03.151Z	DEBUG	netlink/netlink.go:840 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize ADD erspan0(4,erspan) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.151Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:03.152Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:03.157Z	ERROR	netlink/netlink.go:766 (*NetNsProbe).onAddressAdded	jenkins-slave-2.rdocloud: Unable to find interface with index 6 to add address 10.88.0.49/16
2019-04-18T08:10:03.159Z	DEBUG	netlink/netlink.go:569 (*NetNsProbe).addLinkToTopology	jenkins-slave-2.rdocloud: Unable get Bus Info from ethtool (erspan0): operation not supported
2019-04-18T08:10:03.161Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.162Z	DEBUG	netlink/netlink.go:840 (*NetNsProbe).initialize	jenkins-slave-2.rdocloud: Initialize ADD eth0(6,veth) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:03.173Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for cni0(13327,bridge) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.180Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.187Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.194Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.201Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for cni0(13327,bridge) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.207Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:03.227Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:03.227Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:03.439Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:03.555Z	DEBUG	runc/runc.go:262 (*Probe).registerContainer	jenkins-slave-2.rdocloud: Register runc container b1aa81bd3700d7bc6dd8938fe522eeeb106761cac1a4fe58a46b746282f5399d and PID 15713
2019-04-18T08:10:03.555Z	DEBUG	netns/netns.go:134 (*Probe).Register	jenkins-slave-2.rdocloud: Register network namespace: /proc/15713/ns/net
2019-04-18T08:10:03.555Z	DEBUG	netns/netns.go:165 (*Probe).Register	jenkins-slave-2.rdocloud: Increasing counter for namespace 3,4026533337 to 2
2019-04-18T08:10:03.590Z	DEBUG	tests/tests.go:302 execCmds	jenkins-slave-2.rdocloud: Command returned b1aa81bd3700d7bc6dd8938fe522eeeb106761cac1a4fe58a46b746282f5399d

2019-04-18T08:10:03.595Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:04.231Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:04.232Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:04.290Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555574994290, 1555575004290
2019-04-18T08:10:04.447Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:04.807Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:05.630Z	INFO	websocket/client.go:485 (*Client).Connect	jenkins-slave-2.rdocloud: Connecting to ws://127.0.0.1:64500/ws/subscriber/flow
2019-04-18T08:10:05.635Z	DEBUG	websocket/server.go:66 (*Server).serveMessages	jenkins-slave-2.rdocloud: Enforcing websocket for /ws/subscriber/flow, admin
2019-04-18T08:10:05.638Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, websocket, /ws/subscriber/flow ---> true
2019-04-18T08:10:05.639Z	DEBUG	websocket/server.go:78 (*Server).serveMessages	jenkins-slave-2.rdocloud: Serving messages for client jenkins-slave-2.rdocloud for pool /ws/subscriber/flow type : [*websocket.Pool]
2019-04-18T08:10:05.641Z	INFO	websocket/server.go:121 (*Server).newIncomingClient	jenkins-slave-2.rdocloud: New WebSocket Connection from 127.0.0.1:60328 : URI path /ws/subscriber/flow
2019-04-18T08:10:05.641Z	INFO	websocket/client.go:509 (*Client).Connect	jenkins-slave-2.rdocloud: Connected to ws://127.0.0.1:64500/ws/subscriber/flow
2019-04-18T08:10:05.647Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:05.651Z	DEBUG	websocket/pool.go:109 (*incomerPool).AddClient	jenkins-slave-2.rdocloud: AddClient jenkins-slave-2.rdocloud for pool /ws/subscriber/flow type : [*websocket.Pool]
2019-04-18T08:10:05.677Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, injectpacket, write ---> true
2019-04-18T08:10:05.684Z	DEBUG	packetinjector/client.go:267 (*Client).onAPIWatcherEvent	jenkins-slave-2.rdocloud: New watcher event set for 4ac7446a-6ae4-4db3-579d-d27f1ad71ce3
2019-04-18T08:10:05.705Z	DEBUG	packetinjector/inject_packet.go:89 (*PacketInjector).Write	jenkins-slave-2.rdocloud: Injecting packet (98 bytes) on interface cni0
2019-04-18T08:10:05.723Z	DEBUG	packetinjector/client.go:267 (*Client).onAPIWatcherEvent	jenkins-slave-2.rdocloud: New watcher event update for 4ac7446a-6ae4-4db3-579d-d27f1ad71ce3
2019-04-18T08:10:05.743Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:05.766Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:06.719Z	DEBUG	packetinjector/client.go:267 (*Client).onAPIWatcherEvent	jenkins-slave-2.rdocloud: New watcher event delete for 4ac7446a-6ae4-4db3-579d-d27f1ad71ce3
2019-04-18T08:10:06.721Z	INFO	packetinjector/inject_packet.go:195 InjectPackets.func1	jenkins-slave-2.rdocloud: Injection stopped on interface cni0
2019-04-18T08:10:06.770Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:06.965Z	DEBUG	elasticsearch/rollindex.go:83 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Start rolling indices...
2019-04-18T08:10:06.965Z	INFO	elasticsearch/rollindex.go:105 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Index skydive_rolling rolling over
2019-04-18T08:10:06.971Z	INFO	elasticsearch/rollindex.go:120 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Rolling indices done
2019-04-18T08:10:07.775Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:07.831Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:08.171Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555574998171, 1555575008171
2019-04-18T08:10:08.780Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:09.165Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555574999165, 1555575009165
2019-04-18T08:10:09.784Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:10.788Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:11.793Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:11.965Z	DEBUG	elasticsearch/rollindex.go:83 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Start rolling indices...
2019-04-18T08:10:11.965Z	INFO	elasticsearch/rollindex.go:105 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Index skydive_rolling rolling over
2019-04-18T08:10:11.971Z	INFO	elasticsearch/rollindex.go:120 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Rolling indices done
2019-04-18T08:10:12.764Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 0, 1555575012764
2019-04-18T08:10:12.765Z	DEBUG	flow/table.go:292 (*Table).update	jenkins-slave-2.rdocloud: Send updated Flows: 8
2019-04-18T08:10:12.767Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:e61498cb3842b7b6 LayersPath:Ethernet/IPv6/ICMPv6/ICMPv6RouterSolicitation Application:ICMPv6RouterSolicitation Link:A:"d6:0c:9b:74:72:4e" B:"33:33:00:00:00:02"  Network:Protocol:IPV6 A:"fe80::d40c:9bff:fe74:724e" B:"ff02::2"  Transport:<nil> ICMP:Type:ROUTER  DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:1 ABBytes:70 Start:1555575004735 Last:1555575012764  Metric:ABPackets:1 ABBytes:70 Start:1555575004735 Last:1555575004735  TCPMetric:<nil> IPMetric:<nil> Start:1555575004735 Last:1555575004735 RTT:0 TrackingID:4324968c77c6a0d0 L3TrackingID:a87499c7aac76de3 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.768Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:32fb25cccc7a3477 LayersPath:Ethernet/IPv4/ICMPv4 Application:ICMPv4 Link:A:"d6:0c:9b:74:72:4e" B:"00:11:22:33:44:55"  Network:Protocol:IPV4 A:"10.88.0.1" B:"10.88.0.49"  Transport:<nil> ICMP:Type:ECHO  DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:1 ABBytes:98 Start:1555575005705 Last:1555575012764  Metric:ABPackets:1 ABBytes:98 Start:1555575005705 Last:1555575005705  TCPMetric:<nil> IPMetric: Start:1555575005705 Last:1555575005705 RTT:0 TrackingID:46c651f8569b324f L3TrackingID:3468bb468575d699 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.769Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:9ffd854c9524bc36 LayersPath:Ethernet/IPv6/IPv6HopByHop/ICMPv6/MLDv2MulticastListenerReport Application:MLDv2MulticastListenerReport Link:A:"d6:0c:9b:74:72:4e" B:"33:33:00:00:00:16"  Network:Protocol:IPV6 A:"fe80::d40c:9bff:fe74:724e" B:"ff02::16"  Transport:<nil> ICMP: DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:2 ABBytes:260 Start:1555575003112 Last:1555575012764  Metric:ABPackets:2 ABBytes:260 Start:1555575003112 Last:1555575003367  TCPMetric:<nil> IPMetric:<nil> Start:1555575003112 Last:1555575003367 RTT:0 TrackingID:ad061f9b629ab1bd L3TrackingID:034eabdebc71f3b3 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.769Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:0ada26f25f04d769 LayersPath:Ethernet/ARP Application:ARP Link:A:"0a:2e:f8:a5:cb:af" B:"ff:ff:ff:ff:ff:ff"  Network:<nil> Transport:<nil> ICMP:<nil> DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:1 ABBytes:42 Start:1555575003142 Last:1555575012764  Metric:ABPackets:1 ABBytes:42 Start:1555575003142 Last:1555575003142  TCPMetric:<nil> IPMetric:<nil> Start:1555575003142 Last:1555575003142 RTT:0 TrackingID:550636c9831e75cb L3TrackingID:5752b0859882550f ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.770Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:ac3ec439a44acbd4 LayersPath:Ethernet/IPv6/IPv6HopByHop/ICMPv6/MLDv2MulticastListenerReport Application:MLDv2MulticastListenerReport Link:A:"0a:2e:f8:a5:cb:af" B:"33:33:00:00:00:16"  Network:Protocol:IPV6 A:"::" B:"ff02::16"  Transport:<nil> ICMP: DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:1 ABBytes:90 Start:1555575003151 Last:1555575012764  Metric:ABPackets:1 ABBytes:90 Start:1555575003151 Last:1555575003151  TCPMetric:<nil> IPMetric:<nil> Start:1555575003151 Last:1555575003151 RTT:0 TrackingID:4859d2d731dc8bae L3TrackingID:2b2bd095ea116e12 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.771Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:30a8c62879e6a0a1 LayersPath:Ethernet/IPv6/ICMPv6/ICMPv6NeighborSolicitation Application:ICMPv6NeighborSolicitation Link:A:"0a:2e:f8:a5:cb:af" B:"33:33:ff:a5:cb:af"  Network:Protocol:IPV6 A:"::" B:"ff02::1:ffa5:cbaf"  Transport:<nil> ICMP:Type:NEIGHBOR  DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:1 ABBytes:86 Start:1555575003151 Last:1555575012764  Metric:ABPackets:1 ABBytes:86 Start:1555575003151 Last:1555575003151  TCPMetric:<nil> IPMetric:<nil> Start:1555575003151 Last:1555575003151 RTT:0 TrackingID:4a22b284e26d45dd L3TrackingID:35ee6ec2f830474f ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.772Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:e2c8e8582b9996e3 LayersPath:Ethernet/IPv6/IPv6HopByHop/ICMPv6/MLDv2MulticastListenerReport Application:MLDv2MulticastListenerReport Link:A:"0a:2e:f8:a5:cb:af" B:"33:33:00:00:00:16"  Network:Protocol:IPV6 A:"fe80::82e:f8ff:fea5:cbaf" B:"ff02::16"  Transport:<nil> ICMP: DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:2 ABBytes:180 Start:1555575004159 Last:1555575012764  Metric:ABPackets:2 ABBytes:180 Start:1555575004159 Last:1555575004399  TCPMetric:<nil> IPMetric:<nil> Start:1555575004159 Last:1555575004399 RTT:0 TrackingID:4e70425e88cbfa46 L3TrackingID:e477b78754737720 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.772Z	DEBUG	analyzer/flow_server.go:169 (*FlowServerUDPConn).Serve.func1	jenkins-slave-2.rdocloud: New flow from UDP connection: {XXX_state:{lastMetric:<nil> link1stPacket:0 network1stPacket:0 updateVersion:0} UUID:f35d846d155d82b2 LayersPath:Ethernet/IPv6/ICMPv6/ICMPv6RouterSolicitation Application:ICMPv6RouterSolicitation Link:A:"0a:2e:f8:a5:cb:af" B:"33:33:00:00:00:02"  Network:Protocol:IPV6 A:"fe80::82e:f8ff:fea5:cbaf" B:"ff02::2"  Transport:<nil> ICMP:Type:ROUTER  DHCPv4:<nil> DNS:<nil> VRRPv2:<nil> LastUpdateMetric:ABPackets:2 ABBytes:140 Start:1555575004159 Last:1555575012764  Metric:ABPackets:2 ABBytes:140 Start:1555575004159 Last:1555575007807  TCPMetric:<nil> IPMetric:<nil> Start:1555575004159 Last:1555575007807 RTT:0 TrackingID:508e6a0885c47826 L3TrackingID:26f23fd9aef03813 ParentUUID: NodeTID:4037dc08-f384-5a64-4090-657e6dcc26ad LastRawPackets:[] RawPacketsCaptured:0 FinishType:NOT_FINISHED}
2019-04-18T08:10:12.797Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:13.064Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555575003064, 1555575013064
2019-04-18T08:10:13.447Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:13.804Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:14.290Z	DEBUG	flow/table.go:267 (*Table).update	jenkins-slave-2.rdocloud: flow table update: 1555575004290, 1555575014290
2019-04-18T08:10:14.808Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:15.535Z	DEBUG	flow/table.go:397 (*Table).processPacketSeq	jenkins-slave-2.rdocloud: 1 Packets received for capture node 4037dc08-f384-5a64-4090-657e6dcc26ad
2019-04-18T08:10:15.814Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 0, <nil>
2019-04-18T08:10:16.420Z	DEBUG	analyzer/flow_server.go:210 (*FlowServer).storeFlows	jenkins-slave-2.rdocloud: 8 flows stored
2019-04-18T08:10:16.432Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.442Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.450Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.476Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.485Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.587Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.597Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.607Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.618Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, topology, read ---> true
2019-04-18T08:10:16.656Z	INFO	subscriber/client.go:62 (*s3Client).WriteObject	jenkins-slave-2.rdocloud: WriteObject /other/20190418T081016Z/00000000.gz result: <nil>
2019-04-18T08:10:16.821Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 1, <nil>
2019-04-18T08:10:16.826Z	INFO	subscriber/client.go:74 (*s3Client).ReadObject	jenkins-slave-2.rdocloud: ReadObject other/20190418T081016Z/00000000.gz result: <nil>
2019-04-18T08:10:16.832Z	DEBUG	websocket/pool.go:79 (*Pool).OnDisconnected	jenkins-slave-2.rdocloud: OnDisconnected jenkins-slave-2.rdocloud for pool /ws/subscriber/flow type : [*websocket.Pool] 
2019-04-18T08:10:16.832Z	DEBUG	websocket/pool.go:137 (*Pool).RemoveClient	jenkins-slave-2.rdocloud: Successfully removed client jenkins-slave-2.rdocloud for pool /ws/subscriber/flow type : [*websocket.Pool]
2019-04-18T08:10:16.840Z	INFO	subscriber/client.go:127 (*s3Client).ListObjects	jenkins-slave-2.rdocloud: ListObjects result: 1, <nil>
2019-04-18T08:10:16.843Z	INFO	subscriber/client.go:108 (*s3Client).DeleteObject	jenkins-slave-2.rdocloud: DeleteObject other/20190418T081016Z/00000000.gz result: <nil>
2019-04-18T08:10:16.844Z	DEBUG	tests/tests.go:299 execCmds	jenkins-slave-2.rdocloud: Executing command [podman kill secadvtest]
2019-04-18T08:10:16.922Z	DEBUG	tests/tests.go:302 execCmds	jenkins-slave-2.rdocloud: Command returned b1aa81bd3700d7bc6dd8938fe522eeeb106761cac1a4fe58a46b746282f5399d

2019-04-18T08:10:16.923Z	DEBUG	tests/tests.go:299 execCmds	jenkins-slave-2.rdocloud: Executing command [podman rm secadvtest]
2019-04-18T08:10:16.965Z	DEBUG	elasticsearch/rollindex.go:83 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Start rolling indices...
2019-04-18T08:10:16.965Z	INFO	elasticsearch/rollindex.go:105 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Index skydive_rolling rolling over
2019-04-18T08:10:16.970Z	INFO	elasticsearch/rollindex.go:120 (*rollIndexService).roll	jenkins-slave-2.rdocloud: Rolling indices done
2019-04-18T08:10:17.243Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for eth0(6,veth) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:17.295Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.295Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,veth) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.296Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.296Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.296Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.296Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.298Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.303Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.304Z	WARN	netlink/netlink.go:1108 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Failed to get Routes: Link not found
2019-04-18T08:10:17.311Z	DEBUG	netlink/netlink.go:1089 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink DEL event for eth0(6) within 64709471-b980-47d3-4014-6bc0f70fb593
2019-04-18T08:10:17.315Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.315Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for veth60ab16b5(13329,device) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.315Z	DEBUG	netlink/netlink.go:1089 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink DEL event for veth60ab16b5(13329) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.328Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for cni0(13327,bridge) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.333Z	DEBUG	netlink/netlink.go:1089 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink DEL event for veth60ab16b5(13329) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.334Z	DEBUG	netlink/netlink.go:1081 (*NetNsProbe).onMessageAvailable	jenkins-slave-2.rdocloud: Netlink ADD event for cni0(13327,bridge) within e2e0533d-2391-4588-6e8e-8fa033215541
2019-04-18T08:10:17.375Z	DEBUG	netns/netns.go:209 (*Probe).Unregister	jenkins-slave-2.rdocloud: Unregister network Namespace: /var/run/netns/cni-a2f256fe-90f6-f3b6-d616-b6230c4fc4de
2019-04-18T08:10:17.376Z	DEBUG	netns/netns.go:229 (*Probe).Unregister	jenkins-slave-2.rdocloud: Decremented counter for namespace 3,4026533337 to 1
2019-04-18T08:10:17.470Z	DEBUG	netns/netns.go:209 (*Probe).Unregister	jenkins-slave-2.rdocloud: Unregister network Namespace: /proc/15713/ns/net
2019-04-18T08:10:17.472Z	DEBUG	netns/netns.go:234 (*Probe).Unregister	jenkins-slave-2.rdocloud: Network namespace deleted: 3,4026533337
2019-04-18T08:10:17.518Z	DEBUG	tests/tests.go:302 execCmds	jenkins-slave-2.rdocloud: Command returned b1aa81bd3700d7bc6dd8938fe522eeeb106761cac1a4fe58a46b746282f5399d

2019-04-18T08:10:17.521Z	DEBUG	config/rbac.go:49 (*logger).Print	jenkins-slave-2.rdocloud: Request: admin, capture, write ---> true
2019-04-18T08:10:17.526Z	DEBUG	client/client.go:357 (*OnDemandProbeClient).onAPIWatcherEvent	jenkins-slave-2.rdocloud: New watcher event delete for 3cbd8c7b-f5ba-4918-6b1e-aa840860a9ab
--- PASS: Test_SecurityAdvisor (14.81s)
	tests.go:447: Creating captures
	tests.go:459: Executing setup commands
	tests.go:472: Checking captures are correctly set up
	tests.go:526: Executing settle function
	tests.go:541: Executing setup function
	tests.go:663: Running checks
	tests.go:690: Running tear down commands
	tests.go:441: Removing existing captures
2019-04-18T08:10:17.533Z	DEBUG	server/server.go:139 (*OnDemandProbeServer).unregisterProbe	jenkins-slave-2.rdocloud: Attempting to unregister probe on node cni0
2019-04-18T08:10:17.534Z	DEBUG	probes/gopacket.go:333 (*GoPacketProbesHandler).unregisterProbe	jenkins-slave-2.rdocloud: Terminating gopacket capture on 3e08b8c7-0ef5-4fa5-5f57-59becb3c5847
```

hunchback commented:
i mean the content of a file placed on the object store, or a snippet from
that content (if the file is too large).

On Sun, Apr 28, 2019, 15:01 Or Ozeri <notifications@github.com> wrote:

> @hunchback <https://github.com/hunchback> what do you mean by "output" ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skydive-project/skydive/pull/1703#issuecomment-487372660>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAVCAI4QDHAC5QV5SIN46ZTPSWG3LANCNFSM4G6P56KA>
> .
>

orozery commented:
@hunchback ok?

```[{"UUID":"52151fd3488b78ab","LayersPath":"Ethernet/IPv4/ICMPv4","Version":"1.0.8","Status":"STARTED","Network":{"Protocol":"IPV4","A":"10.88.0.1","B":"10.88.0.49","B_Name":"0_0_myhost_0"},"LastUpdateMetric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521912796,"RTT":22000},"Metric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521911501,"RTT":22000},"Start":1556521853142,"Last":1556521911501,"UpdateCount":0,"NodeType":"veth"},{"UUID":"e9e1b274207505a7","LayersPath":"Ethernet/ARP","Version":"1.0.8","Status":"STARTED","LastUpdateMetric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521912796},"Metric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521901409},"Start":1556521874320,"Last":1556521901409,"UpdateCount":0,"NodeType":"veth"},{"UUID":"9c279c93d782abba","LayersPath":"Ethernet/IPv4/ICMPv4","Version":"1.0.8","Status":"STARTED","Network":{"Protocol":"IPV4","A":"10.88.0.1","B":"10.88.0.49","B_Name":"0_0_myhost_0"},"LastUpdateMetric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521912822,"RTT":33000},"Metric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521911501,"RTT":33000},"Start":1556521853142,"Last":1556521911501,"UpdateCount":0,"NodeType":"bridge"},{"UUID":"c0a1664d11472550","LayersPath":"Ethernet/ARP","Version":"1.0.8","Status":"STARTED","LastUpdateMetric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521912822},"Metric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521901409},"Start":1556521874320,"Last":1556521901409,"UpdateCount":0,"NodeType":"bridge"},{"UUID":"882721b12bc96196","LayersPath":"Ethernet/IPv4/ICMPv4","Version":"1.0.8","Status":"STARTED","Network":{"Protocol":"IPV4","A":"10.88.0.1","B":"10.88.0.49","B_Name":"0_0_myhost_0"},"LastUpdateMetric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521912828,"RTT":26000},"Metric":{"ABPackets":59,"ABBytes":5782,"BAPackets":59,"BABytes":5782,"Start":1556521853142,"Last":1556521911501,"RTT":26000},"Start":1556521853142,"Last":1556521911501,"UpdateCount":0,"NodeType":"veth"},{"UUID":"e047c039da2078e5","LayersPath":"Ethernet/ARP","Version":"1.0.8","Status":"STARTED","LastUpdateMetric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521912828},"Metric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521874320,"Last":1556521901409},"Start":1556521874320,"Last":1556521901409,"UpdateCount":0,"NodeType":"veth"},{"UUID":"52151fd3488b78ab","LayersPath":"Ethernet/IPv4/ICMPv4","Version":"1.0.8","Status":"UPDATED","Network":{"Protocol":"IPV4","A":"10.88.0.1","B":"10.88.0.49","B_Name":"0_0_myhost_0"},"LastUpdateMetric":{"ABPackets":60,"ABBytes":5880,"BAPackets":60,"BABytes":5880,"Start":1556521912796,"Last":1556521972796,"RTT":17000},"Metric":{"ABPackets":119,"ABBytes":11662,"BAPackets":119,"BABytes":11662,"Start":1556521853142,"Last":1556521971640,"RTT":17000},"Start":1556521853142,"Last":1556521971640,"UpdateCount":1,"NodeType":"veth"},{"UUID":"e9e1b274207505a7","LayersPath":"Ethernet/ARP","Version":"1.0.8","Status":"UPDATED","LastUpdateMetric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521912796,"Last":1556521972796},"Metric":{"ABPackets":4,"ABBytes":168,"BAPackets":4,"BABytes":168,"Start":1556521874320,"Last":1556521955553},"Start":1556521874320,"Last":1556521955553,"UpdateCount":1,"NodeType":"veth"},{"UUID":"9c279c93d782abba","LayersPath":"Ethernet/IPv4/ICMPv4","Version":"1.0.8","Status":"UPDATED","Network":{"Protocol":"IPV4","A":"10.88.0.1","B":"10.88.0.49","B_Name":"0_0_myhost_0"},"LastUpdateMetric":{"ABPackets":60,"ABBytes":5880,"BAPackets":60,"BABytes":5880,"Start":1556521912822,"Last":1556521972821,"RTT":26000},"Metric":{"ABPackets":119,"ABBytes":11662,"BAPackets":119,"BABytes":11662,"Start":1556521853142,"Last":1556521971640,"RTT":26000},"Start":1556521853142,"Last":1556521971640,"UpdateCount":1,"NodeType":"bridge"},{"UUID":"c0a1664d11472550","LayersPath":"Ethernet/ARP","Version":"1.0.8","Status":"UPDATED","LastUpdateMetric":{"ABPackets":2,"ABBytes":84,"BAPackets":2,"BABytes":84,"Start":1556521912822,"Last":1556521972821},"Metric":{"ABPackets":4,"ABBytes":168,"BAPackets":4,"BABytes":168,"Start":1556521874320,"Last":1556521955553},"Start":1556521874320,"Last":1556521955553,"UpdateCount":1,"NodeType":"bridge"}]```

hunchback commented:
@orozery - thanks for the file content - its' fine
